torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -642,6 +642,208 @@ class IrParser {
+            auto M = constant_as<int>(node->input(5));
+            TORCH_INTERNAL_ASSERT(
+                M.has_value(), "The M parameter is required.");
+            const size_t kBatchSize = M.value();
+
+            // N = product of [reduction_axis, input_ndims]
+            // Repurposed for NvFuser such that N = norm_shape_ndims
+            // so we can construct reduction_axes and broadcast_mask
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            auto N = constant_as<int>(node->input(6));
+            TORCH_INTERNAL_ASSERT(
+                N.has_value(), "The N parameter is required.");
+            const size_t kNormShapeNumDims = N.value();
+
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            auto out_mask_list = constant_as<c10::List<bool>>(node->input(7));
+            TORCH_INTERNAL_ASSERT(
+                out_mask_list.has_value(),
+                "output mask for layer_norm_backward");
+            std::vector<int> output_mask;
+            for (const auto value : out_mask_list->vec()) {
+              output_mask.emplace_back(static_cast<int>(value));
+            }
+
+            const size_t kOuterNumDims = input->nDims() - kNormShapeNumDims;
+
+            std::vector<int> outer_reduction_axes(kOuterNumDims);
+            std::vector<bool> outer_broadcast_mask(input->nDims(), false);
+            for (size_t idx = 0; idx < kOuterNumDims; ++idx) {
+              outer_reduction_axes[idx] = idx;
+              outer_broadcast_mask[idx] = true;
+            }
+
+            std::vector<int> inner_reduction_axes(kNormShapeNumDims);
+            std::vector<bool> inner_broadcast_mask(input->nDims(), false);
+            Val* num_features = nullptr;
+            for (int idx = 0; idx < kNormShapeNumDims; ++idx) {
+              const size_t axis = input->nDims() - 1 - idx;
+              inner_reduction_axes[idx] = axis;
+              inner_broadcast_mask[axis] = true;
+              num_features = (num_features == nullptr)
+                  ? input->domain()->domain()[axis]->extent()
+                  : mul(num_features,
+                        input->domain()->domain()[axis]->extent());
+            }
+
+            // TODO: grad_bias and grad_weight are disabled because
+            // they are incompabilble with grad_in fusion
+            // Requires seperate kernels
+
+            /*
+            auto grad_bias = sum(grad_out, outer_reduction_axes);
+            fusion.addOutput(grad_bias);
+
+            auto bcast_mean = broadcast(mean, inner_broadcast_mask);
+            auto bcast_rstd = broadcast(rstd, inner_broadcast_mask);
+            auto x_hat = mul(sub(input, bcast_mean), bcast_rstd);
+            auto grad_weight = sum(mul(grad_out, x_hat), outer_reduction_axes);
+            fusion.addOutput(grad_weight);
+            */
+
+            // auto bcast_mean = broadcast(mean, inner_broadcast_mask);
```

Comment:
nitpick: dead code.

Reply:
This is the code to generate the weight and bias gradients. It needs a second kernel.